src/Packages.ts

```diff
@@ -0,0 +1,66 @@
+import * as fs from 'fs'
+import path from 'path'
+
+import { Descriptor } from './Descriptor'
+import { LsifSymbol } from './LsifSymbol'
+
+export class Packages {
+  constructor(public readonly projectRoot: string) {}
+  private cache: Map<string, LsifSymbol> = new Map()
+  public symbol(filePath: string): LsifSymbol {
+    if (path.normalize(filePath) !== filePath) {
+      throw new Error('BOOM FIX THIS')
+    }
+    // TODO: handle end case.
+    const fromCache = this.cache.get(filePath)
+    if (fromCache) {
+      return fromCache
+    }
+    const packageJsonPath = path.join(filePath, 'package.json')
+    try {
+      if (
+        // eslint-disable-next-line no-sync
```

Comment:
Should we turn off this lint altogether? It seems needlessly tedious to turn it off every time you want to do a synchronous I/O operation.